docs/clustering/multi-node-setup.rst

```diff
@@ -319,10 +319,14 @@ the quorum size using the `discovery.zen.minimum_master_nodes`_ setting.
 
 .. NOTE::
 
-    For a three-node cluster, CrateDB will set the quorum size to three.
-    Consequentially, you must declare all nodes to be master-eligible. Consult
-    the `quorum guide`_ for detailed information about quorum size
-    considerations.
+    CrateDB will set the quorum size to ``FLOOR(n / 2) + 1``, so that is the 
+    minimum number of master-eligible nodes you should have.
+    For high availability purposes you want to have one additional
```

Comment:
> For high availability purposes you want to have one additional master-eligible node on top of the minimum required for quorum for each node loss you want to tolerate.

It might be just me, but I was having trouble understanding this part until I re-read it several times.
How about something like:
> For high availability, increase the number of master-eligible nodes by one for each node loss you want to compensate.

Reply:
Doing further tests I found that the requirement for a quorum of data nodes does not seem to extend to the number of master-eligible nodes, it seems sufficient to have one master-eligible node among the nodes that are available. No warning seems to be triggered when the number of master eligible nodes is below "quorum level", so I removed the entire section from this document.

Reply:
Looking further I believe nodes that are not master-eligible still carry a copy of the cluster state, so at that point I think it makes sense to only check for a quorum of data-carrying nodes.

Reply:
Yes right, the quorum part is only relevant for CrateDB < 4.x. Just to clarify and not relevant to this change (but more to your comment), on a fresh new cluster, `initial_master_nodes` must be set to > 1 as these are the nodes which are allowed to vote at the first ever master node election. If only 1 node would be listed here, election wouldn't work. But as noted this is only needed on a fresh cluster (and in production), after the first election has taken place, this setting is not needed anymore and is even recommended to be removed.

Reply:
Thank you.
My thinking removing the section in the latest revision in this branch is to make the how-to more readable for people looking at new deployments which would be with versions >=4.0.0

Regarding `initial_master_nodes` listing only 1 node, for new 5.1.2 clusters, I have confirmed this works both 
- for a 3 nodes cluster (the first time the node that is listed gets elected as master, but in subsequent cold starts another node gets elected without issues if the node that was listed for the first election is not available)
- and for a 1-node cluster (`discovery.seed_hosts` set to the hostname of the node - `recover_after_data_nodes`=1 - `expected_data_nodes`=1), later scaled to 2 and then 3 nodes. Tried both individual node restarts and full stop and start of just 2 nodes and master election seems to work fine.

Reply:
Did you try that out on nodes not bound the loopback interface? If bound to the loopback interface the cluster is treat as in-development and thus master election will work even without any `initial_master_nodes` set.

Reply:
As said, this is only related to the very first master election of a fresh cluster. It is irrelevant for nodes joining an existing cluster or node restarts.

Reply:
We are mixing again, high availability (which depends on the replication settings) and general availability (which depends on a quorum in the clusters voting configuration).

and seemingly also `initial_master_nodes` and `node.master: true`

>  CrateDB will set the quorum size to ``FLOOR(n / 2) + 1```

with `n` being the number of nodes in the voting configuration (!) not the total number of master eligible nodes.
`FLOOR(4/2) + 1 == 3` for a 4 node cluster the quorum however is 2.

The voting config is calculated with:
```java
size - (size % 2 == 0 ? 1 : 0);
```

3 master eligible nodes -> 3 nodes in voting config -> quorum reached with 2
4 master eligible nodes -> 3 nodes in voting config -> quorum reached with 2
5 master eligible nodes -> 5 nodes in voting config -> quorum reached with 3
5 master eligible nodes -> 6 nodes in voting config -> quorum reached with 4

---

> 5-nodes cluster can tolerate the loss of 2 nodes and so on

The cluster itself can tolerate the **simultaneous** loss of 2 nodes.
However you can perfectly fine remove up to 3 nodes from a 5 node cluster for it still to work (as cluster), which does not imply, that all data is available. You just need to wait till the voting configurations adapt to the new number of nodes.

> Doing further tests I found that the requirement for a quorum of data nodes does not seem to extend to the number of master-eligible nodes, it seems sufficient to have one master-eligible node among the nodes that are available

Could you be more explicit what you identified here and maybe post the relevant settings.